cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_nodegroup_test.go

```diff
@@ -753,125 +753,91 @@ func TestNodeGroupDecreaseSize(t *testing.T) {
 	}
 }
 
-func TestNodeGroupMachineSetDeleteNodes(t *testing.T) {
-	// Note: 10 is an upper bound for this test. Going beyond 10
-	// will break the sorting that happens later in this function
-	// because sort.Strings() will not do natural sorting and the
-	// expected semantics in this test will fail.
-	nodes := make([]*apiv1.Node, 10)
-	machines := make([]*v1beta1.Machine, 10)
-	nodeObjects := make([]runtime.Object, 10)
-	machineObjects := make([]runtime.Object, 10)
+func TestNodeGroupDeleteNodes(t *testing.T) {
+	test := func(t *testing.T, testObjs *clusterTestConfig) {
+		t.Helper()
 
-	machineSet := &v1beta1.MachineSet{
-		TypeMeta: v1.TypeMeta{
-			Kind: "MachineSet",
-		},
-		ObjectMeta: v1.ObjectMeta{
-			Name:      "machineset",
-			Namespace: "test-namespace",
-			UID:       "abcdef12-a3d5-a45f-887b-6b49aa8fc218",
-			Annotations: map[string]string{
-				nodeGroupMinSizeAnnotationKey: "1",
-				nodeGroupMaxSizeAnnotationKey: "3",
-			},
-		},
-		Spec: v1beta1.MachineSetSpec{
-			Replicas: int32ptr(int32(len(machines))),
-		},
-	}
+		controller, stop := testObjs.newMachineController(t)
+		defer stop()
 
-	for i := 0; i < len(nodes); i++ {
-		nodes[i] = &apiv1.Node{
-			TypeMeta: v1.TypeMeta{
-				Kind: "Node",
-			},
-			ObjectMeta: v1.ObjectMeta{
-				Name: fmt.Sprintf("node-%d", i),
-				Annotations: map[string]string{
-					machineAnnotationKey: fmt.Sprintf("test-namespace/machine-%d", i),
-				},
-			},
-			Spec: apiv1.NodeSpec{
-				ProviderID: fmt.Sprintf("providerid-%d", i),
-			},
+		ng, err := testObjs.newNodeGroup(t, controller)
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
 		}
 
-		machines[i] = &v1beta1.Machine{
-			TypeMeta: v1.TypeMeta{
-				Kind: "Machine",
-			},
-			ObjectMeta: v1.ObjectMeta{
-				Name:      fmt.Sprintf("machine-%d", i),
-				Namespace: "test-namespace",
-				OwnerReferences: []v1.OwnerReference{{
-					Name: machineSet.Name,
-					Kind: machineSet.Kind,
-					UID:  machineSet.UID,
-				}},
-			},
-			Status: v1beta1.MachineStatus{
-				NodeRef: &apiv1.ObjectReference{
-					Kind: nodes[i].Kind,
-					Name: nodes[i].Name,
-				},
-			},
+		nodeNames, err := ng.Nodes()
```

Comment:
Is `len(nodeNames) > 0` ever? I see a new node is created by calling `testObjs.newNodeGroup` which initializes an empty node group. When/Where are the nodes actually added to the node group?

Reply:
In the call to `newMachineSetTestObjs()` where `t.Run()` is called for each test case.

Reply:
I see, `newMachineSetTestObjs(t.Name(), 0, 10, 10,` compared to `newMachineSetTestObjs(namespace string, id, nodeCount int, replicaCount int32,`. Ok, 10 -> 10 nodes.